server/src/models/teamDistribution.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  Entity,
+  Column,
+  PrimaryGeneratedColumn,
+  CreateDateColumn,
+  UpdateDateColumn,
+  Index,
+  ManyToOne,
+  OneToMany,
+  ManyToMany,
+} from 'typeorm';
+import { Course } from './course';
+import { CourseTask } from './courseTask';
+import { Student } from './student';
+import { Team } from './team';
+
+@Entity()
+export class TeamDistribution {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @CreateDateColumn()
+  createdDate: Date;
+
+  @UpdateDateColumn()
+  updatedDate: Date;
+
+  @ManyToOne(_ => Course, { nullable: true })
+  course: Course;
+
+  @Column({ nullable: true })
+  @Index()
+  courseId: number;
+
+  @Column({ type: 'timestamptz', nullable: true })
+  distributionStartDate: null | Date;
+
+  @Column({ type: 'timestamptz', nullable: true })
+  distributionEndDate: null | Date;
+
+  @Column()
+  name: string;
+
+  @Column({ nullable: true })
+  description: string;
+
+  @OneToMany(_ => CourseTask, courseTask => courseTask.teamDistribution)
+  courseTasks: CourseTask[];
+
+  @ManyToMany(_ => Student, student => student.teamDistribution)
+  students: Student[];
```

Comment:
why do we need it?  we have `teamDistribution` -> `team` -> `student` relation, don't we?

Reply:
We need this because students will immediately register for distribution and they won't have teams.

Reply:
It probably won't work that way. You need join table for many-to-many